### ISSUE TITLE
Merge with PowerPoint file on ubuntu doesn't work

### ISSUE THREAD

**Issue by madina972** [albertjuhe/camaleonX#2]
Hi,
I tried your script for merging few pptx file and it doesn't work.

My camaleonx.properties file : 

```
base=/home/myusername/camaleonx/resources/files/
results=/home/myusername/camaleonx/results/

#xsl docx path
app=/home/myusername/camaleonx/resources/
tmp=/home/myusername/camaleonx/resources/tmp/
xsl=xsl/
skeleton=skeleton/

tmp_pptx=tmp/
skeleton_pptx=skeleton_pptx/
xsl_pptx=xsl_pptx/

#Templates
templates=templates/
```

Identical to your camalonx.properties.

I can see that the directory "skeleton_pptx" not exist, is that normal ?
Only skeleton/, xsl/, xsl_pptx/ folder exist.

My pptx_merge_sample.xml file : 

```
<?xml version="1.0" encoding="UTF-8" ?>
<profile>
    <metas>
        <meta key="project_name" value="pptx_sample"/>
        <meta key="type" value="pptx"/> <!-- docx, pptx -->
    </metas>
    <contents>
        <content include="1" href="1.pptx"/>
        <content  include="1" href="2.pptx"/>
        <content include="1" href="3.pptx"/>
    </contents>
</profile>
```

When I run the script, i'm getting this :

```
Reading config file: camaleonx.properties
CamaleonX v.0.1 Command line execution.
Processing pptx_merge_sample.xml...
log4j:WARN No appenders could be found for logger (omaonk.xml.profile).
log4j:WARN Please initialize the log4j system properly.
Start merging [PPTX]...
[Pptx pptx_merge_sample] Number of documents to merge: 3
Start merging... 
Current document: 1
Merging: /home/myusername/camaleonx/1.pptx
Directory does not exist: /home/myusername/camaleonx/resources/tmp/pptx_merge_sample/result/
Current document: 2
Merging: /home/myusername/camaleonx/2.pptx
Directory does not exist: /home/myusername/camaleonx/resources/tmp/pptx_merge_sample/result/
Saving result file /home/myusername/camaleonx/resources/tmp/pptx_merge_sample/file1 as /home/myusername/camaleonx/results/...
Result file saved OK.
Temporal files removed.
MERGE TIME:00h 00m 00s 117ms, 2 Documents.
End meging PPTX FilesPPTX'.
```

This will create a .pptx file however only the first pptx file slide's will be inside.

For this result :

```
Directory does not exist: /home/myusername/camaleonx/resources/tmp/pptx_merge_sample/result/
```

If I create by myself the tmp/pptx_merge_sample/result folder, I'm not getting this error for the first document but still getting it for the second one and the 3rd document is not event handled.

**Comment by albertjuhe** [comment]
Hi,

I ve been checking the pptx mergin in my localhost and works.
I've seen that the folder skeleton_pptx doesn't exits in the Github but is in my localhost project, this could be the problem, now I'm trying to upload this folder but Github don't detect the existenxe.

Ive added too, five pptx that I use as a Test.

**Comment by Cirrusware** [comment]
Hello,

I have the same issue.

I create missing folder with this command
mkdir -p ./resources/tmp/pptx_merge_sample/result/

I launch this
java -jar camaleonx.jar -U camaleonx.properties.ubuntu pptx_merge_sample.xml -Xms1024m -Xmx1024m -Dlog4j.configuration=file:/./camaleonx.log4j.xml -Dlog4j.debug=true -Dcamaleonx.local.home=logs
Result :
mkdir: impossible de créer le répertoire «./resources/skeleton_pptx»: Le fichier existe
Reading config file: camaleonx.properties.ubuntu
CamaleonX v.0.1 Command line execution.
Processing pptx_merge_sample.xml...
log4j:WARN No appenders could be found for logger (omaonk.xml.profile).
log4j:WARN Please initialize the log4j system properly.
Start merging [PPTX]...
[Pptx pptx_merge_sample] Number of documents to merge: 5
Start merging... 
Current document: 1
Merging: /home/download/camaleonX-master/./documents/1.pptx
Current document: 2
Merging: /home/download/camaleonX-master/./documents/2.pptx

Application stop, without merge.

Have you an idea?

---

I have tested too docx merge, I have this result
Reading config file: camaleonx.properties.ubuntu
CamaleonX v.0.1 Command line execution.
Processing docx_merge_sample.xml...
log4j:WARN No appenders could be found for logger (omaonk.xml.profile).
log4j:WARN Please initialize the log4j system properly.
Start merging [Docx]...
[Docx docx_merge_sample] Number of documents to merge: 2
[Docx docx_merge_sample]  Start merging... 
[Docx docx_merge_sample] Current document number: 1
[Docx docx_merge_sample] Merging: /home/download/camaleonX-master/./documents/1.docx
[Docx docx_merge_sample] Merge OK.
[Docx docx_merge_sample] Current document number: 2
[Docx docx_merge_sample] Merging: /home/download/camaleonX-master/./documents/2.docx
Recoverable error on line 83 of mergeDocx.xsl:
  FODC0002: I/O error reported by XML parser processing
  file:/home/download/camaleonX-master/resources/xsl/resources/resources/tmp/docx_merge_sample/file2/word/document.xml: /home/download/camaleonX-master/resources/xsl/resources/resources/tmp/docx_merge_sample/file2/word/document.xml (Aucun fichier ou dossier de ce type)
Recoverable error on line 50 of mergeStyles.xsl:
  FODC0002: I/O error reported by XML parser processing
  file:/home/download/camaleonX-master/resources/xsl/resources/resources/tmp/docx_merge_sample/file2/word/styles.xml: /home/download/camaleonX-master/resources/xsl/resources/resources/tmp/docx_merge_sample/file2/word/styles.xml (Aucun fichier ou dossier de ce type)
Recoverable error on line 29 of mergeStyles.xsl:
  FODC0002: I/O error reported by XML parser processing
  file:/home/download/camaleonX-master/resources/xsl/resources/resources/tmp/docx_merge_sample/file1/word/numbering.xml: /home/download/camaleonX-master/resources/xsl/resources/resources/tmp/docx_merge_sample/file1/word/numbering.xml (Aucun fichier ou dossier de ce type)
Recoverable error on line 80 of index.xsl:
  FODC0002: I/O error reported by XML parser processing
  file:/home/download/camaleonX-master/resources/xsl/resources/resources/tmp/docx_merge_sample/file2/Content_Types.xml: /home/download/camaleonX-master/resources/xsl/resources/resources/tmp/docx_merge_sample/file2/Content_Types.xml (Aucun fichier ou dossier de ce type)
Recoverable error on line 88 of index.xsl:
  FODC0002: Document has been marked not available:
  file:/home/download/camaleonX-master/resources/xsl/resources/resources/tmp/docx_merge_sample/file2/Content_Types.xml
[Docx docx_merge_sample] Merge OK.
[Docx docx_merge_sample] Merged all files Correctly.
[Docx docx_merge_sample] Removing temporal files./home/download/camaleonX-master/./resources/./resources/tmp/docx_merge_sample
[Docx docx_merge_sample] Merge Report:  
[Docx docx_merge_sample] Total merge time:00h 00m 00s 953ms, 2 documents merged.
[Docx docx_merge_sample] Merge ended DOCX'.
[Docx docx_merge_sample] Result file in: ./results/testdocx_2016_06_26_08_15_04.docx

**Comment by madina972** [comment]
Hi albertjuhe,

Did you test on Linux ?

**Comment by albertjuhe** [comment]
Yes, I have a virtualbox with ubuntu and I tested in this environment, let me try again and I wiill informe you about it.

**Comment by rvalery** [comment]
Hello albertjuhe,

Any updates on this issue ? Are you able to merge the pptx on ubuntu ? I can run the doc example, but pptx gives me the same result as Cirrusware (temporary files seems to be created in resources/tmp, but the script just stops without any information of errors)

Thank you,
Raphaël Valéry

**Comment by irama** [comment]
Hey albertjuhe, very promising codebase. Thank you for your incredible efforts. But having same issue as above on MacOSX. Any news?
